website_snippet_product_category/views/product_public_category.xml

```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <data>
+        <record id="product_public_category_form_view" model="ir.ui.view">
+            <field name="name">public.category.form</field>
+            <field name="model">product.public.category</field>
+            <field
+                name="inherit_id"
+                ref="website_sale.product_public_category_form_view"
+            />
+            <field name="arch" type="xml">
+                <field name="sequence" position="after">
+                    <field name="website_published" />
```

Comment:
Why not putting this as smart-button? Any way, the default label for the field is not very descriptive for this use case, so I think you need to override that string.